src/diffusers/models/transformers/transformer_2d.py

```diff
@@ -153,91 +150,267 @@ def __init__(
                 f" {patch_size}. Make sure that `in_channels`, `num_vector_embeds` or `num_patches` is not None."
             )
 
-        # 2. Define input layers
-        if self.is_input_continuous:
-            self.in_channels = in_channels
+        self.in_channels = in_channels
 
-            self.norm = torch.nn.GroupNorm(num_groups=norm_num_groups, num_channels=in_channels, eps=1e-6, affine=True)
-            if use_linear_projection:
-                self.proj_in = linear_cls(in_channels, inner_dim)
-            else:
-                self.proj_in = conv_cls(in_channels, inner_dim, kernel_size=1, stride=1, padding=0)
+        # 2. Initialize the right blocks.
+        if self.is_input_continuous:
+            self._init_continuous_input(
+                in_channels,
+                out_channels,
+                inner_dim,
+                num_attention_heads,
+                attention_head_dim,
+                cross_attention_dim,
+                activation_fn,
+                num_embeds_ada_norm,
+                attention_bias,
+                only_cross_attention,
+                double_self_attention,
+                upcast_attention,
+                norm_type,
+                norm_elementwise_affine,
+                norm_eps,
+                attention_type,
+                norm_num_groups,
+                use_linear_projection,
+                dropout,
+                num_layers,
```

Comment:
I don't think all the args need to be passed here no? Think we can reduce to just these. They should be enough to configure the relevant attributes. 

```python
in_channels
norm_num_groups
use_linear_projection
sample_size
num_vector_embeds
inner_dim
interpolation_scale
norm_type
```